app/assets/javascripts/discourse/tests/acceptance/user-menu-test.js

```diff
@@ -102,31 +104,49 @@ acceptance("User menu", function (needs) {
 
     assert.strictEqual(
       customTab1.dataset.tabNumber,
-      "5",
+      "6",
       "custom tab has the right tab number"
     );
 
     assert.strictEqual(
       customTab2.dataset.tabNumber,
-      "6",
+      "7",
       "custom tab has the right tab number"
     );
 
     const reviewQueueTab = query("#user-menu-button-review-queue");
 
     assert.strictEqual(
       reviewQueueTab.dataset.tabNumber,
-      "7",
+      "8",
       "review queue tab comes after the custom tabs"
     );
 
-    const tabs = [...queryAll(".tabs-list .btn")]; // top and bottom tabs
+    const tabs = [...queryAll(".tabs-list.top-tabs .btn")];
 
     assert.deepEqual(
-      tabs.map((t) => t.dataset.tabNumber),
-      ["0", "1", "2", "3", "4", "5", "6", "7", "8"],
+      tabs.reduce((acc, tab) => {
+        acc[tab.id] = tab.dataset.tabNumber;
+        return acc;
+      }, {}),
+      {
+        "user-menu-button-all-notifications": "0",
+        "user-menu-button-replies": "1",
+        "user-menu-button-mentions": "2",
+        "user-menu-button-likes": "3",
+        "user-menu-button-messages": "4",
+        "user-menu-button-bookmarks": "5",
+        "user-menu-button-custom-tab-1": "6",
+        "user-menu-button-custom-tab-2": "7",
+        "user-menu-button-review-queue": "8",
+      },
```

Comment:
I was initially thinking to extract this to the top of the file since the numbers technically don't change. Then L105, L111, L119 can use the map too.

😅

Reply:
Don't really want to block you on my opinions, though.

Reply:
Nope, this is totally fair. I've moved the map to the top of the test (instead of the file) since it's specific to this test case that adds custom tabs. I've also removed the duplicate custom tabs data-tab-number assertions.

Reply:
Thanks a lot Osama